### PR TITLE
Feat/database config and connection#34

### PR DESCRIPTION
**Pull request by mcdonnell88**
includes...
-room model and room schema
-user model and user schema
-database config and connection
-also includes a bunch of ugly commits -- trying to resolve merge conflicts and fix stupid mistakes (originally forgot to branch, but now resolved)

### CODE REVIEW

File: server/users/userModel.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  name: {
+    first: String,
+    last: String
+  },
+  email: String,
+  date_created: {
+    type: Date,
+    default: Date.now
+  },
+  profile_photo: ??,
```
Comment (garronmichael) [inline server/users/userModel.js]: I would store this as a string for now.

Comment (mcdonnell88) [inline server/users/userModel.js]: ah thanks.  Forgot I had that '??' line in there.

File: server/app.js [hunk 2]
```diff
@@ -7,4 +7,19 @@ var port = process.env.port || 3000;
...
 
-app.listen(port);
\ No newline at end of file
+app.listen(port);
+
+
+
+
+// INITIATION: include mongoose and open connection to database on locally running MongoDB instance
+var mongoose = require('mongoose');
```
Comment (puopg) [inline server/app.js]: Please move this up to the top, we should keep these kinds of dependencies all in one spot in a given file

File: server/app.js [hunk 3]
```diff
@@ -7,4 +7,19 @@ var port = process.env.port || 3000;
...
+mongoose.connect(<MONGODB connection goes here>) --> example: 'mongodb://localhost/test'
+
+
+// CONNECTION: pending to database on local host; need notification if connected successfully or error occured
+var db = mongoose.connection;
+db.on('error', console.error.bind(console, 'connection error:'));
+db.once('open', function(callback){
+  console.log('connection is open!');
```
Comment (puopg) [inline server/app.js]: Can you change this comment to "MongoDB connection open!"

Comment (mcdonnell88) [inline server/app.js]: got it, thanks!

**Comment by mcdonnell88** [comment]
updated pull request with comments.  Thanks!

**Pull request merged**